### PR TITLE
improve: don't cache test campaigns

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
There's no point in caching test campaigns.

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by donnie-jp — APPROVED** [review]
LGTM

**Review by malayli — APPROVED** [review]

File: RInAppMessaging/Classes/RInAppMessaging.swift [hunk 1]
```diff
@@ -123,7 +123,6 @@ import Foundation
     /// - Parameter provider: object that will always contain up-to-date user information.
     @objc public static func registerPreference(_ provider: UserInfoProvider) {
         inAppQueue.async {
-            notifyIfModuleNotInitialized()
```
Comment (vincentdelmundo) [inline RInAppMessaging/Classes/RInAppMessaging.swift:126]: Just want to confirm, why is this removed? Is this because `registerPreference` can be set before `configure`?

**Review by vincentdelmundo — APPROVED** [review]
Looks good. Just one to confirm 1 thing, please see inline comment.

File: RInAppMessaging/Classes/RInAppMessaging.swift (hunk 1 above)
Comment (imaginaris) [reply RInAppMessaging/Classes/RInAppMessaging.swift:126]: That's right. I think it's better to let clients set `UserInfoProvider` even before init/configure to avoid any issues with API calls order. Only `logEvent` requires `configure` to be called before.
